fastlane/Fastfile

```diff
@@ -433,16 +491,45 @@ platform :ios do
   #
   desc 'Trigger a new beta build on CI'
   lane :new_beta_release do |options|
-    ios_betabuild_prechecks(options)
+    # Verify that there's nothing in progress in the working copy
+    ensure_git_status_clean
 
-    # Re-generate the strings for GlotPress, just in case there were
-    # localization fixes.
+    # Check out the up-to-date default branch, the designated starting point for the code freeze
+    Fastlane::Helper::GitHelper.checkout_and_pull(DEFAULT_BRANCH)
```

Comment:
Why do we need to check and pull `trunk` when releasing a new beta?

Reply:
Ah, good question. Thank you @leandroalonso 

From top of memory, the previous implementation of the version bump would look for the value on `trunk` to determine which new beta version to make. Therefore, we needed beta 1 merged into `trunk` to start beta 2, or we'd have to specify the new beta version manually.

@spencertransier might be able to give more details on whether the new process still requires that. If not, we might as well avoid the dance.

Reply:
I think it's not a bad thing to have, just wanted to know. :)

Reply:
@leandroalonso @mokagio The new tooling does not need to check `trunk` out first, so I replaced this line with a `release/` branch check instead: https://github.com/Automattic/pocket-casts-ios/commit/89f08f295c64e6cc5bdfecb49a053f30356ff07b

Reply:
@spencertransier This is going to be a bit too much but I think it gives enough detail to try to understand what happened:

1. I released the beta `7.52.0.1`
2. I then realized #1240 was merged into `trunk`
3. We then opened #1243 
4. In the meantime, `7.52.0.1` was ready and #1245 was open and was waiting for CI build to pass
5. I then started the new beta, but it said the new beta would be `7.52.0.1` (it should have been `7.52.0.2`)
6. I waited till #1245 was merged into `trunk`
7. Tried again and this time it said the new beta would be `7.52.0.2`, like it should

You can reproduce this right now if you want:

1. #1247 is not merged
2. Go to `release/7.53` and then `bundle exec fastlane new_beta_release`
3. What is expected: the next beta to be `7.52.0.3`, instead it will say it will be `7.52.0.2`

That's why I think we need to update `trunk`, otherwise it might affect the beta version?

Reply:
Ohhhhhh got it!

Thanks for the explanation @spencertransier!

Reply:
@leandroalonso You're welcome! 😃 Are you good with me merging this PR now?